src/components/lib/component/arch/x86_64/cos_asm_simple_stacks.h

```diff
@@ -25,6 +25,57 @@
 
 #define COS_ASM_REQUEST_STACK
 
+#define COS_ULINV_SWITCH_DOMAIN(protdom)			\
```

Comment:
Protdom is still JITed. This macro is just to make the assembly more readable because we do this operation multiple times.

Reply:
> I'm a little surprised to see the protdom argument here as I figured most of the arguments would come from the jit rewriting.

Protdom is still JITed. This macro is just to make the assembly more readable because we do this operation multiple times.